richreview/public/webapps/richreview/r2/controller/controller_event.js

```diff
@@ -9,6 +9,73 @@ var r2Ctrl = {};
 /** @namespace r2 */
 (function(r2){
 
+    /* cookie helper functions */
+    r2.setCookie = (function(cname, cvalue, exdays) {
+        var d = new Date();
+        d.setTime(d.getTime() + (exdays*24*60*60*1000));
+        var expires = "expires="+d.toUTCString();
+        document.cookie = cname + "=" + cvalue + "; " + expires;
+    });
+
+    r2.getCookie = (function(cname) {
+        var name = cname + "=";
+        var ca = document.cookie.split(';');
+        for(var i=0; i<ca.length; i++) {
+            var c = ca[i];
+            while (c.charAt(0)==' ') c = c.substring(1);
+            if (c.indexOf(name) == 0) return c.substring(name.length,c.length);
+        }
+         return "";
+    });
+
+    /*
+    r2.drag = (function() {
+        console.log("enter r2.drag");
+        var dragging = false;
+
+        $('#dragbar').mousedown(function(e) {
+            //e.preventDefault();
+            console.log("drag bar mouse down");
+            dragging = true;
+        });
+
+        $('#dragbar').mouseup(function(e) {
+            if (dragging) {
+                console.log("drag resize");
+                var app_container_size = r2.viewCtrl.getAppContainerSize();
+                var left_percentage = (e.pageX / (app_container_size - 40)) * 100;
+                var right_percentage = 100 - left_percentage;
+                $('#r2_view').css("width", left_percentage + "%");
+                $('#r2_thumbnail').css("width", right_percentage + "%");
+                dragging = false;
+            }
+        });
+
+    }());
+    */
```

Comment:
Please create a singleton that wraps r2.dragging, r2.dragStart, and r2.dragEnd inside. You will use them like:
r2.previewDragBar.Start(), and r2.previewDragBar.End()